### PR TITLE
Added game state variable

### PR DESCRIPTION
**Pull request by PBWiseman**
This branch adds a game state variable that can be either InGame or Paused. I think I have found the places in the code this needs to be added but I am not sure I caught all of them.

### CODE REVIEW

File: Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -7,21 +7,12 @@ public class GameManager : MonoBehaviour
 {
     public GameObject gameOverUI;
     public ScoreManager scoreManager;
-   private bool playerDead = false;
```
Comment (Loerad) [inline Assets/Scripts/GameManager.cs:10]: I think this section should be under a different game state as this causes the item screen bug

File: Assets/Scripts/Player/TopDownMovement.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ void Start()
 
     void Update()
     {
+        if (GameSettings.gameState != GameState.InGame){return;}
```
Comment (Loerad) [inline Assets/Scripts/Player/TopDownMovement.cs:25]: Should stop all movement as well when the game isn't running.

**Review by Loerad — CHANGES_REQUESTED** [review]
Code looks really easy to implement if we want to add it ourselves to new additions we make. Couple bugs: After the item menu pops up, movement isn't stopped if you are still holding down a movement key. When the player dies the item menu appears and the player can pick items.

File: Assets/Scripts/GameManager.cs (hunk 1 above)
Comment (PBWiseman) [reply Assets/Scripts/GameManager.cs:10]: ItemSelect enum added

File: Assets/Scripts/Player/TopDownMovement.cs (hunk 2 above)
Comment (PBWiseman) [reply Assets/Scripts/Player/TopDownMovement.cs]: Fixed

**Review by kblack89 — CHANGES_REQUESTED** [review]
The code makes sense but when playing I found that apon death you cannot choose main menu , putting in your score works and reset makes the game reset but no movement can happen.

**Comment by PBWiseman** [comment]
> The code makes sense but when playing I found that apon death you cannot choose main menu , putting in your score works and reset makes the game reset but no movement can happen.

Issue fixed

**Review by kblack89 — APPROVED** [review]
After playing through these issues are resolved, code makes sense and no errors or bugs.

File: Assets/Scripts/Inventory/InventoryController.cs [hunk 3]
```diff
@@ -6,20 +6,18 @@ public class InventoryController : MonoBehaviour
 {
     //this is the controller for the inventory panel's control.
     public GameObject timerManager;
-    public Timer timer;
     [SerializeField]
     private InventoryPage inventoryUI;
     [HideInInspector]
     public int inventorySize = 3;
 
     public void Start()
```
Comment (AlexReidNZ) [inline Assets/Scripts/Inventory/InventoryController.cs:14]: This empty void could be deleted.

**Review by AlexReidNZ — APPROVED** [review]
This code functions as intended.

File: Assets/Scripts/Inventory/InventoryController.cs (hunk 3 above)
Comment (PBWiseman) [reply Assets/Scripts/Inventory/InventoryController.cs]: Done

**Pull request merged**